src/DssDirectDepositHub.sol

```diff
@@ -0,0 +1,465 @@
+            }
+        }
+
+        if (amount == 0 && fees == 0) {
+            emit Unwind(ilk, 0);
+            return;
+        }
+
+        require(amount <= 2 ** 255, "DssDirectDepositHub/overflow");
+
+        // To save gas you can bring the fees back with the unwind
+        uint256 total = _add(amount, fees);
+        pool.withdraw(total);
+        daiJoin.join(address(this), total);
+
+        // normalized debt == erc20 DAI to pool (Vat rate for this ilk fixed to 1 RAY)
+
+        if (mode == Mode.NORMAL) {
+            vat.frob(ilk, address(pool), address(pool), address(this), -int256(amount), -int256(amount));
+            vat.slip(ilk, address(pool), -int256(amount));
+            vat.move(address(this), vow, _mul(fees, RAY));
+        } else if (mode == Mode.MODULE_CULLED) {
+            vat.slip(ilk, address(pool), -int256(amount));
+            vat.move(address(this), vow, _mul(total, RAY));
+        } else {
+            // This can be done with the assumption that the price of 1 aDai equals 1 DAI.
+            // That way we know that the prev End.skim call kept its gap[ilk] emptied as the CDP was always collateralized.
+            // Otherwise we couldn't just simply take away the collateral from the End module as the next line will be doing.
+            vat.slip(ilk, address(end_), -int256(amount));
+            vat.move(address(this), vow, _mul(total, RAY));
+        }
+
+        emit Unwind(ilk, amount);
+    }
+
+    function exec(bytes32 ilk_) external {
+        D3MPoolLike pool = ilks[ilk_].pool;
+
+        uint256 availableAssets = pool.maxWithdraw();
+
+        if (vat.live() == 0) {
+            uint256 culled = ilks[ilk_].culled;
+            // MCD caged
+            require(end.debt() == 0, "DssDirectDepositHub/end-debt-already-set");
+            require(culled == 0, "DssDirectDepositHub/module-has-to-be-unculled-first");
+            _unwind(
+                ilk_,
+                pool,
+                type(uint256).max,
+                availableAssets,
+                Mode.MCD_CAGED
+            );
+        } else if (live == 0) {
+            uint256 culled = ilks[ilk_].culled;
+            // This module caged
+            _unwind(
+                ilk_,
+                pool,
+                type(uint256).max,
+                availableAssets,
+                culled == 1
+                ? Mode.MODULE_CULLED
+                : Mode.NORMAL
+            );
+        } else {
+            // Normal path
+            (uint256 totalAssets, uint256 targetAssets) = ilks[ilk_].plan.calcSupplies(availableAssets);
+
+            if (targetAssets > totalAssets) {
+                _wind(ilk_, pool, targetAssets - totalAssets);
+            } else if (targetAssets < totalAssets) {
+                _unwind(
+                    ilk_,
+                    pool,
+                    totalAssets - targetAssets,
+                    availableAssets,
+                    Mode.NORMAL
+                );
+            }
```

Comment:
So plan should still be updated to return the target position size instead of this Aave-specific `calcSupplies` with concepts of `availableLiquidity`, `supplyAmount`, etc.

Code can be updated to this after the change is made as we are only interested in the delta and not the absolute value:

```
uint256 targetPositionSize = plan.getTargetAssets();
uint256 currPositionSize = pool.assetBalance();

if (targetPositionSize > currPositionSize) {
    _wind(ilk_, pool, targetPositionSize - currPositionSize);
} else if (targetPositionSize < currPositionSize) {
    _unwind(
        ilk_,
        pool,
        currPositionSize - targetPositionSize,
        availableAssets,
        Mode.NORMAL
    );
}
```

Reply:
@talbaneth do you see any issue with this on the compound side?

Reply:
No, this is fine.

Reply:
https://github.com/makerdao/dss-direct-deposit/pull/31/commits/184dc19589ff1fb50d4db0288a588cd85eceb1a5